server/pulp/plugins/rsync/publish.py

```diff
@@ -378,30 +364,11 @@ def is_fastforward(self):
         :rtype: bool
         """
         force_full = False
-        for entry in self.predist_history:
-            predistributor_force_full = entry.get("distributor_config", {}).get("force_full",
-                                                                                False)
-            force_full |= predistributor_force_full
-            if entry.get("result", "error") == "error":
```

Comment:
i might agree on the fact that probably we do not need to rsync in a force_full mode if publish of the predistributor failed.